Automattic/wp-calypso
# Add Do Not Sell Dialog component to public pages in wp-calypso

#### Proposed Changes

- Add DoNotSell dialog to universal footer components (for wpcom and jpcom)
- Improve tracking prefs setter
- Add unified useDoNotSell hook
- Improve DoNotSell privacy setting to depend on common logic

#### Testing Instructions (Calypso)

* Run `yarn start` (calypso)
* Open url `http://calypso.localhost:3000/plugins?flags=cookie-banner` (as logged out)
* Set cookies for CCPA region: `country_code=US; region=California;`
* Reload the page and scroll to the bottom
* You should see "Do Not Sell or Share My Data" link
![CleanShot 2022-12-09 at 11 56 44@2x](https://user-images.githubusercontent.com/8419292/206763896-326eedea-e01c-4e37-a577-45eaa75174c8.png)

* Click on the link, Do Not Sell dialog should open
<img width="758" alt="CleanShot 2022-12-09 at 11 57 52@2x" src="https://user-images.githubusercontent.com/8419292/206763963-4d382ef9-1060-4aa5-955e-04e76ffcf71b.png">


* Clicking the toggle button should change `advertising` bucket in cookie `sensitive_pixel_options`
* You should be able to close the dialog either by clicking outside of it, "x" button, or "Close" button
* Opening dialog should not scroll the viewport to the top of the page
* Set your cookies to `country_code=NL;`
* Reload, scroll to the bottom and "Do Not Sell or Share My Data" should not be visible
<img width="1728" alt="CleanShot 2022-12-09 at 11 58 41@2x" src="https://user-images.githubusercontent.com/8419292/206764072-29ea34d0-2e45-4c0d-a3f1-7dd35a5acac3.png">


#### Testing Instructions (Jetpack Cloud)

* Run `yarn start-jetpack-cloud-p` (calypso)
* Open url `http://jetpack.cloud.localhost:3001/pricing?flags=cookie-banner` (as logged out)
* Set cookies for CCPA region: `country_code=US; region=California;`
* Reload the page and scroll to the bottom
* You should see "Do Not Sell or Share My Data" link
![CleanShot 2022-12-09 at 12 01 53@2x](https://user-images.githubusercontent.com/8419292/206764673-36122b7a-0266-413b-b396-d9e7e3dc849e.png)

* Click on the link, Do Not Sell dialog should open
<img width="726" alt="CleanShot 2022-12-09 at 12 02 21@2x" src="https://user-images.githubusercontent.com/8419292/206764729-e7dcd525-157d-47f4-ae25-3f0cfa15742e.png">

* Clicking the toggle button should change `advertising` bucket in cookie `sensitive_pixel_options`
* You should be able to close the dialog either by clicking outside of it, "x" button, or "Close" button
* Opening dialog should not scroll the viewport to the top of the page
* Set your cookies to `country_code=NL;`
* Reload, scroll to the bottom and "Do Not Sell or Share My Data" should not be visible
<img width="1728" alt="CleanShot 2022-12-09 at 12 01 28@2x" src="https://user-images.githubusercontent.com/8419292/206764569-31f9b270-cba3-4a16-9add-5eede70c93d3.png">


#### Pre-merge Checklist





Related to #70682 and #70552
